solutions/bio1/BlockChains/index.tsx

```diff
@@ -0,0 +1,58 @@
+﻿import { SCodeBlock, SText, STitle, SList } from "components";
+//import { brace, floor } from "solutions/util";
+
+export const BlockChains = () => {
+    return (
+        <>
+            <STitle>Observation</STitle>
+            <SText>
+                A straightforward way to solve this problem is to generate all possible permutations of the blocks and then filter out those that violate the no-three-in-order condition.
+                Specifically, we would:
+            </SText>
+            <SList>
+                <SText> 1. Generate all possible permutations of the remaining letters that could follow the prefix p.</SText>
+                <SText> 2. Check each permutation to see if it contains any three letters in alphabetical order. </SText>
+                <SText> 3. Count and return the number of permutations that are valid. </SText>
+            </SList>
+            <SText>
+                However, this approach quickly becomes too slow as l increases.
+                One way to optimise is generate permutations letter by letter, and discard permutations as soon as the no-three-in-order condition is broken.
+                Again however, this is not enough for a full solution.
+            </SText><SText>
+                The key insight is that when constructing permutations, we only care about the sequences of letters in alphabetical order in the prefix, and of those, for each length of sequence,
+                the one with the lexicographically smallest ending letter. We will store the ending letters of such sequences of lengths 1 and 2 as (a, b), where a is the lexicographically smallest
```

Comment:
surround all numbers in dollar signs, and I think you should write (a, b) as `$(a, b)$`  please

same thing for `$a$`